hazelcast/src/test/java/com/hazelcast/test/compatibility/CompatibilityTestUtils.java

```diff
@@ -32,25 +28,18 @@
     // loaded with final modifier to allow subclass proxying.
     public static void attachFinalRemovalAgent() {
         Instrumentation instrumentation = ByteBuddyAgent.install();
-        new AgentBuilder.Default()
-                .disableClassFormatChanges()
-                .type(new ElementMatcher<TypeDescription>() {
-                    @Override
-                    public boolean matches(TypeDescription target) {
-                        return target.getName().startsWith("com.hazelcast");
-                    }
-                })
-                .transform(new AgentBuilder.Transformer() {
-                    @Override
-                    public DynamicType.Builder<?> transform(DynamicType.Builder<?> builder,
-                                                            TypeDescription typeDescription,
-                                                            ClassLoader classLoader, JavaModule module) {
-                        int actualModifiers = typeDescription.getActualModifiers(false);
-                        // unset final modifier
-                        int nonFinalModifiers = actualModifiers & ~Opcodes.ACC_FINAL;
-                        return builder.modifiers(nonFinalModifiers);
-                    }
-                })
-                .installOn(instrumentation);
+        new AgentBuilder.Default().with(AgentBuilder.TypeStrategy.Default.REDEFINE)
+                                  .with(AgentBuilder.InitializationStrategy.NoOp.INSTANCE)
```

Comment:
Previously used `disableClassFormatChanges()` was actually disallowing removal of `final` modifier, silently disregarding the transformer. We configure the agent with `REDEFINE` type strategy and `NoOp` initialization strategy; this allows for the redefinition of the type (instead of default `REBASE` strategy) without adding any methods (which result in modifying `Serializable` classes' `serialVersionUid`). For more details see [`ByteBuddy#rebase`](https://github.com/raphw/byte-buddy/blob/master/byte-buddy-dep/src/main/java/net/bytebuddy/ByteBuddy.java#L775) vs [`ByteBuddy#redefine`](https://github.com/raphw/byte-buddy/blob/master/byte-buddy-dep/src/main/java/net/bytebuddy/ByteBuddy.java#L697).

Reply:
+1 for the `REDEFINE`/`REBASE` background info, what about adding it as a code comment for posterity?